client/config/config.go

```diff
@@ -485,6 +508,58 @@ func (c *ClientTemplateConfig) IsEmpty() bool {
 		c.NomadRetry.IsEmpty()
 }
 
+func (c *ClientTemplateConfig) Merge(o *ClientTemplateConfig) *ClientTemplateConfig {
+	if c == nil {
+		return o
+	}
+
+	result := *c
+	if o == nil {
+		return &result
+	}
+
+	if len(o.FunctionDenylist) > 0 {
+		funcSet := set.From(result.FunctionDenylist)
+		funcSet.InsertSlice(o.FunctionDenylist)
+		result.FunctionDenylist = funcSet.Slice()
+	}
+
+	if len(o.FunctionBlacklist) > 0 {
+		funcSet := set.From(result.FunctionBlacklist)
+		funcSet.InsertSlice(o.FunctionBlacklist)
+		result.FunctionBlacklist = funcSet.Slice()
+	}
```

Comment:
Should we just set `result.FunctionDenylist = slices.Copy(o.FunctionDenylist)`? Making merge the union makes it impossible to allow default denied funcs.

Reply:
Good catch, will fix.

Reply:
Done in [3296220](https://github.com/hashicorp/nomad/pull/20165/commits/3296220865af93364468ba0160695abb63948da3) along with a test exercising the difference in behavior between `function_deny_list = []` and being unset.